sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -196,9 +205,15 @@ def fit(self, X, y=None):
         X = check_array(X, dtype=[X_DTYPE], force_all_finite=False)
         max_bins = self.n_bins - 1
 
+        sample_weight_bin = np.ones(X.shape[0])
```

Comment:
I would rename this to `subsampling_probabilities` to make it explicit that those weights are only used when `subsample` is large enough to trigger resampling. Let's move that below the `if self.subsample is not None and X.shape[0] > self.subsample` condition.